smart-contracts/wasm-chain-integration/src/v1/types.rs

```diff
@@ -445,6 +446,7 @@ impl Output for ImportFunc {
                 ReceiveOnlyFunc::GetReceiveEntrypointSize => 29,
                 ReceiveOnlyFunc::GetReceiveEntryPoint => 30,
                 ReceiveOnlyFunc::Invoke => 31,
+                ReceiveOnlyFunc::Upgrade => 32,
```

Comment:
I don't think you can use 32, it is already used for 
```
ImportFunc::Common(CommonFunc::VerifyEd25519)
```

Also you need to add this tag to the
```
    fn parse(
        ctx: Ctx,
        cursor: &mut std::io::Cursor<&'a [u8]>,
    ) -> wasm_transform::parse::ParseResult<Self> {
```
method.